### ISSUE TITLE
WCAG 1.3.1 test 4 false positives and false negatives

### ISSUE THREAD

**Issue by jzuidweg** [veraPDF/veraPDF-wcag-algs#220]
# False positives

WCAG 1.3.1 test 4 generates errors for almost _all_ formatted text, which is probably too sensitive.

Examples:
- In [test-document 15.pdf](https://github.com/veraPDF/veraPDF-wcag-algs/files/9110020/test-document-15.pdf) and [test document 16](https://github.com/veraPDF/veraPDF-wcag-algs/files/9109996/test-document-16.pdf), it generates errors for list bullets.
- In [test-document-14.pdf](https://github.com/veraPDF/veraPDF-wcag-algs/files/9110106/test-document-14.pdf), it generates errors for superscripts of footnotes.
- In [test document 3](https://github.com/veraPDF/veraPDF-wcag-algs/files/9110152/test-document-3.pdf) and all above test documents, it generates errors for single words or short lines set in boldface or italics. 

# False negatives

In [test document 3](https://github.com/veraPDF/veraPDF-wcag-algs/files/9110152/test-document-3.pdf), the algorithm should probably call out the texts which have been highlighted in yellow, but it doesn't.

# About this test

This is probably a very difficult test to implement because formatting text is in itself not prohibited by WCAG 2.1. It becomes a problem when the document contains information that is represented **only by formatting**. 

Example: for the following text the algorithm should generate errors, because italics are used to convey real information that cannot be seen in any other way:

> In the following paragraph, verbs are marked in italics: Congress _shall make_ no law _respecting_ an establishment of religion, or _prohibiting_ the free exercise thereof; or _abridging_ the freedom of speech, or of the press; or the right of the people peaceably to _assemble_, and to _petition_ the Government for a redress of grievances.

Example:  for the following text the algorithm should not generate errors, because italics are only used to emphasize text, but not for conveying new information:
> The first amendment: Congress shall make no law respecting an _establishment of religion_, or _prohibiting the free exercise thereof_; or _abridging the freedom of speech_, or of the press; or the _right of the people peaceably to assemble_, and to petition the Government for a redress of grievances.

It's very difficult for an algorithm to distinguish between these cases, because it has to work out whether the formatting is used to convey new information that is not otherwise visible. This probably requires AI, which is out of the scope of this issue. But perhaps as a patch we can use a few simple heuristic rules:

- **Boldface** and _italics_ are mostly used for just emphasizing text, so we should probably not generate errors for this.
- List bullets should not generate errors.
- Table cells which are empty but have a different background colour should generate errors, because there is a big chance that colours of empty table cells are used to convey information.
- Highlighted text (such as in [test document 3](https://github.com/veraPDF/veraPDF-wcag-algs/files/9110152/test-document-3.pdf)) and coloured text should generate errors because colouring is (as far as I know) not read out by screen readers.
